operator/v1/types_ingress.go

```diff
@@ -1399,6 +1399,132 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported via another mechanism in Ingress.Spec.RequiredHSTSPolicies.
+	// The header settings made here can replace settings made in: cache-control, spec.clientTLS,
+	// spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId,
+	// and spec.httpHeaders.headerNameCaseAdjustments.
+	// The header settings made here can be overwritten by spec.httpHeaders or annotations
+	// on Route objects.
+	// Setting httpCaptureHeaders for the headers set via this API will not work.
+	// Setting "cookie" or "set-cookie" headers that conflict with OpenShift router's session cookies will result in undefined behavior.
```

Comment:
Really? Is that a behaviour that we want? I feel like this should be resolved. Should we be allowing cookies to be set via the actions API?

Reply:
I Would you prefer a separate API for configuring cookies?

Reply:
Well I guess I would be interested to know why you think someone would want to use the `set-cookie` header in this API before we make a decision, are there use cases you can think of? Do you think it would be restrictive if we banned the use of the `set-cookie` header here?

Perhaps only to forbid the cookie names we know set by the router is possible?

I just feel like we should either prevent users doing something bad here or understand what kind of things will happen if they overwrite these headers

Also, are there any security implication in them rewriting cookie headers related to sessions?

Reply:
I'm sure use-cases for configuring cookies exist, but I don't have a ready list of use-cases; none were requested in [RFE-464](https://issues.redhat.com/browse/RFE-464).  Maybe it could be useful to delete problematic cookies for buggy applications, or to add a cookie to indicate which route was used to connect to the application.  Anyway, I would be fine with prohibiting cookies entirely until such a time as we get an RFE for configuring cookies, at which time we could define a structured API for the same.  

The cookies that HAProxy sets are used for session tracking, to map a client connection to a particular backend server.  The cookie name is an MD5 hash value derived from the backend's name (similarly, the cookie value is a hashed value derived from backend server's endpoint).  So the cookie names that we would want to forbid are all the possible values of the MD5 hashing algorithm, but it's extremely unlikely that someone would _accidentally_ choose the same cookie name that HAProxy used.

Reply:
> at which time we could define a structured API for the same.

A structured API or relaxing the validation that prevents cookies here would be acceptable. I think at the moment I'm just uncomfortable with the "undefined behaviour" part of this, it's like we haven't explored it to understand the issue enough. 
If we think it needs investigation, but want to defer that, I'm ok blocking cookies for now and adding support, as you say, when we get an RFE for it

Reply:
for now blocked it till we get an api for it.

Reply:
@JoelSpeed I had added the block header list here in the go doc which you suggested here https://github.com/openshift/api/pull/1364/commits/4c22439926bd0e58754e72adfc43cc3ce6340d4f#r1275191144